src/editors/containers/TextEditor/components/ImageSettingsModal/hooks.test.js

```diff
@@ -0,0 +1,314 @@
+import { StrictDict } from '../../../../utils';
+import * as hooks from './hooks';
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useEffect: jest.fn(),
+}));
+
+const simpleDims = { width: 3, height: 4 };
+const reducedDims = { width: 7, height: 13 };
+const gcd = 7;
+const multiDims = {
+  width: reducedDims.width * gcd,
+  height: reducedDims.height * gcd,
+};
+
+const stateKeys = StrictDict(Object.keys(hooks.state).reduce(
+  (obj, key) => ({ ...obj, [key]: key }),
+  {},
+));
+
+const hookKeys = StrictDict(Object.keys(hooks).reduce(
+  (obj, key) => ({ ...obj, [key]: key }),
+  {},
+));
+
+let oldState;
+const setState = {};
+const mockState = () => {
+  oldState = hooks.state;
+  const keys = Object.keys(stateKeys);
+  keys.forEach(key => {
+    setState[key] = jest.fn();
+    hooks.state[key] = jest.fn(val => [val, setState[key]]);
```

Comment:
This step is critical, in that we are monkeypatching in our own value for hooks.state for each key, and because we are doing it multiple times, we don't mock at the module level? Just trying to clarify my understanding, so that I can identify places to use this pattern because it seems neat.